checker/src/org/checkerframework/checker/lock/LockAnnotatedTypeFactory.java

```diff
@@ -96,28 +107,310 @@ public LockTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, LockS
         return new LockTransfer((LockAnalysis) analysis,(LockChecker)this.checker);
     }
 
-    protected AnnotatedTypeMirror getDeclaredAndDefaultedAnnotatedType(Tree tree) {
-        shouldCache = false;
+    class LockQualifierHierarchy extends MultiGraphQualifierHierarchy {
+
+        public LockQualifierHierarchy(MultiGraphFactory f) {
+            super(f, LOCKHELD);
+        }
+
+
+        boolean isGuardedBy(AnnotationMirror am) {
+            return AnnotationUtils.areSameIgnoringValues(am, GUARDEDBY) ||
+                   AnnotationUtils.areSameIgnoringValues(am, JAVAXGUARDEDBY) ||
```

Comment:
Last time I tried it, aliasing didn't make the three annotations identical as far as subtype checks are concerned. Is it supposed to?
Before I refactor the code - are you in favor of removing support for the jcip and javax annotations?
This is potentially something we could revisit (i.e. we can remove support now for this release, and consider adding it later).